### PR TITLE
Update useParticipants to listen for any Participant's info changes

### PR DESCRIPTION
**Pull request by lebaudantoine**
### Issue
This is my first contribution! I’d love any advice on how to contribute to the repo. While working with the `useParticipants` hook, I expected it to keep the participants list updated with their names. However, I found that `useParticipants` only listens to metadata changes by default and does not react to updates in names or attributes.

To ensure the participants list updates correctly, I called the hook as follows:

```javascript
const participants = useParticipants({
  updateOnlyOn: [
    RoomEvent.ParticipantNameChanged,
    ...allParticipantRoomEvents,
  ],
});
```

Additionally, I noticed that the term `updateOnlyOn` could be misleading, as it always includes three events. While the documentation clarifies this, it may cause confusion when reading the code.

### Solution

I found the previous implementation a bit convoluted. I propose that `useParticipants` should update on all room events that trigger state changes for a participant, as originally intended by @Ocupe. Additionally, please note that a few events related to end-to-end encryption and permissions are still missing from the current setup.

### CODE REVIEW

**Comment by lebaudantoine** [comment]
what do you think @lukasIO?

**Review by lukasIO — APPROVED** [review]

File: .changeset/calm-rivers-behave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@livekit/components-core': minor
```
Comment (lukasIO) [inline .changeset/calm-rivers-behave.md:2]:
```suggestion
'@livekit/components-core': patch
```

as it's primarily a bugfix, I think we should mark it as a patch update

**Comment by lukasIO** [comment]
@lebaudantoine feel free to open a subsequent PRs for the other missing events, we'd definitely want them to be part of `allRemoteParticipantEvents` as well!

File: .changeset/calm-rivers-behave.md (hunk 1 above)
Comment (lebaudantoine) [reply .changeset/calm-rivers-behave.md]: Fixed, ready for merge!

**Pull request merged**